compiler/next/include/chpl/uast/Conditional.h

```diff
@@ -0,0 +1,206 @@
+/*
+ * Copyright 2021 Hewlett Packard Enterprise Development LP
+ * Other additional copyright holders may be indicated within.
+ *
+ * The entirety of this work is licensed under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef CHPL_UAST_CONDITIONAL_H
+#define CHPL_UAST_CONDITIONAL_H
+
+#include "chpl/queries/Location.h"
+#include "chpl/uast/BlockStyle.h"
+#include "chpl/uast/ControlFlow.h"
+
+namespace chpl {
+namespace uast {
+
+
+/**
+  This class represents a conditional. For example:
+
+  \rst
+  .. code-block:: chapel
+
+      // Example 1:
+      const flag = false;
+      if flag then writeln('a'); else writeln('b');
+
+  \endrst
+
+ */
+class Conditional final : public ControlFlow {
+ private:
+  Conditional(ASTList children, int8_t conditionChildNum,
+              BlockStyle thenBlockStyle,
+              int8_t thenBodyChildNum,
+              int numThenBodyStmts,
+              BlockStyle elseBlockStyle,
+              int elseBodyChildNum,
+              int numElseBodyStmts,
+              bool isExpressionLevel)
```

Comment:
Conditionals can also appear at an expression level:

```chapel
var x = if 3 > 2 then 'a' else 'b';
```

The same goes for some kinds of `IndexableLoop` (e.g. `forall`, `for`).

Reply:
Got it. Out of curiosity, if the condition is implicit can the block style be explicit. 

`var x = if 3 > 2 then { tmp = foo(); return tmp + 'a' } else { tmp = foo(); return tmp + 'b' };

Reply:
I'm not sure what you mean - are you differentiating between `if 3 > 2` and `if (3 > 2)` ? Anyway in your example that should not compile because it has statements in it but the `var x = ....` can only have an expression.

Reply:
Sorry, I meant to say if the conditionals appears at an expression level can the block style be explicit.

That is can you have multiline bodies for  "then" and "else" if isExpressionLevel == true?

Reply:
no

Reply:
In the case we may want to add something to the effect of: assert(isExpressionLevel == false || (thenBlockStyle != EXPLICIT && elseBlockStyle != EXPLICIT))

Reply:
I think I have individual asserts in there which amount to the same sort of check, let me...check... :D